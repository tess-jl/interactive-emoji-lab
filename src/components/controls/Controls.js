import React from 'react';
import PropTypes from 'prop-types';
import styles from './Controls.css';

const Controls = ({ actions }) => (
  <section className={styles.Controls}>
    {actions.map(({ name, text, count, actionCreator }) => (
      <button key={name} onClick={actionCreator}>
        {text || name} {!!count && `- ${count}`}
      </button>
    ))}
  </section>
);

Controls.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    text: PropTypes.string
  })).isRequired
};

export default Controls;

