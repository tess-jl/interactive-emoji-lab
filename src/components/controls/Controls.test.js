import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls';

describe('Controls component', () => {
  let wrapper;
  beforeEach(() => {
    const actions = [
      { name: 'DO_STUFF', text: 'stuff' },
      { name: 'other' }
    ];

    wrapper = shallow(<Controls
      actions={actions}
    />);
  });

  it('renders Controls', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
  
