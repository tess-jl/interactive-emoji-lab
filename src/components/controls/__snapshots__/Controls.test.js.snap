// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders Controls 1`] = `
<section
  className="Controls"
>
  <button
    key="DO_STUFF"
  >
    stuff
     
  </button>
  <button
    key="other"
  >
    other
     
  </button>
</section>
`;
